run_dir/static/js/workset_samples.js

```diff
@@ -0,0 +1,80 @@
+/*
+File: workset_samples.js
+URL: /static/js/workset_samples.js
+Powers /workset/[workset] - template is run_dir/design/workset_samples.html
+*/
+
+// Get data attributes
+var workset_name= $('#workset-js').attr('data-workset');
+
+
+$.getJSON("/api/v1/workset/"+workset_name, function(data) {
+    
+    // Fill in the main table with summary information
+    var content="";
+    var wsdata= data[workset_name];
+    var tbody = '<tr> \
+                     <th>Project id</th> \
+                     <th>Project name</th> \
+                     <th>Application</th> \
+                     <th>Library method</th> \
+                     <th>Samples</th> \
+                     <th>Library info</th> \
+                     <th>Sequencing info</th> \
+                 </tr> ';
+    
+    if(wsdata.hasOwnProperty("projects")){
+        $.each(wsdata.projects, function(project_id, project_data){
+            content+='<h2>Project <a href="/project/'+project_id+'">'+project_id+'</a></h2>';
+            content+='<table class="table table-bordered narrow-headers" id="ws-'+project_id+'"> \
+                     <tr> \
+                     <th>Project name</th> \
+                     <th>Application</th> \
+                     <th>Library method</th> \
+                 </tr> ';
+            var samplesnb=Object.keys(project_data['samples']).length;
+            content+="<tr> \
+                    <td>"+project_data['name']+"</td>\
+                    <td>"+project_data['application']+"</td>\
+                    <td>"+project_data['library']+"</td></tr></table>";
+        content+="<h3>Samples</h3>";
+        content+='<table class="table table-bordered narrow-headers" id="ws-'+project_id+'"> \
+                     <tr> \
+                     <th>Sample name</th> \
+                     <th>Library</th> \
+                     <th>Sequencing</th> \
+                 </tr> ';
+        $.each(project_data.samples, function(sample_id, sample_data){
+        content+="<tr><td>"+sample_id+"</td> \
+                <td>";
+            $.each(sample_data.library, function(lib_id, lib_data){
+                lims_id=lib_id.split("-")[1];
+                content+="<a href='https://genologics.scilifelab.se:8443/clarity/work-complete/"+lims_id+"'>"+lib_id+"</a> - "+lib_data['date']+" - "+lib_data['status']+"<br />";
```

Comment:
Not really. I guess that the nicest way would be to write a new API call in genstat instead of calling the LIMS directly, though might not be worth the effort.. The URL just caught my eye :wink: